src/main/java/com/ray/message/controller/MessageController.java

```diff
@@ -0,0 +1,62 @@
+package com.ray.message.controller;
+
+import com.ray.message.annotations.NeedAuth;
+import com.ray.message.exception.ServiceException;
+import com.ray.message.model.Message;
+import com.ray.message.rep.PaginatedResponse;
+import com.ray.message.rep.RespResult;
+import com.ray.message.sevice.MessageService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Date;
+import java.util.List;
+
+@RestController
+@RequestMapping("/message")
+public class MessageController {
+
+    @Autowired
+    private MessageService messageService;
+
+    @GetMapping("/messages")
+    @NeedAuth
+    public RespResult getMessageRange(Integer thisUserId, @RequestParam(value = "otherUserId") Integer otherUserId,
+                                      @RequestParam(value = "startTime") Date startTime,
+                                      @RequestParam(value = "endTime") Date endTime,
+                                      @RequestParam(value = "pageSize") int pageSize,
+                                      @RequestParam(value = "page") int page) throws ServiceException {
+        List<Message> messagesByTimeRange = messageService.getMessagesByTimeRange(thisUserId, otherUserId, startTime, endTime, pageSize, page);
+        boolean hasMore = messageService.countByTimeRange(thisUserId, otherUserId, startTime, endTime) > (page - 1) * pageSize + messagesByTimeRange.size();
+        return RespResult.createWithData(new PaginatedResponse<Message>(page, messagesByTimeRange, hasMore));
+    }
+
+    @PostMapping("/send")
+    @NeedAuth
+    public RespResult send(Integer userId, @RequestBody Message message) {
+        messageService.addMessage(userId, message);
+        return RespResult.createWithData("sent successfully");
+    }
+
+    @GetMapping("/new_messages")
```

Comment:
Consider keep consistent on the URL namings for all APIs: either `abc_xyz` or `abcXyz`